### PR TITLE
WFLY-17123 Drop source transformation for jpa subsystem

### PR DESCRIPTION
**Pull request by scottmarlow**
Signed-off-by: Scott Marlow <smarlow@redhat.com>
https://issues.redhat.com/browse/WFLY-17123

I can build the wildfly/jpa folder with these changes but trying to build all of WildFly gets failure:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project wildfly-ejb3: Compilation failure: Compilation failure: 
[ERROR] /home/smarlow/work/wildfly/ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionBeanClassTable.java:[35,25] package javax.persistence does not exist
[ERROR] /home/smarlow/work/wildfly/ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionBeanClassTable.java:[36,25] package javax.persistence does not exist
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :wildfly-ejb3
```

CC @chengfang

### CODE REVIEW

**Comment by chengfang** [comment]
@scottmarlow  in legacy-ee pom.xml, can we keep the legacy.jakarta.persistence-api? This is used referenced by ejb class `org.jboss.as.ejb3.component.stateful.StatefulSessionBeanClassTable`

**Comment by scottmarlow** [comment]
> @scottmarlow in legacy-ee pom.xml, can we keep the legacy.jakarta.persistence-api? This is used referenced by ejb class `org.jboss.as.ejb3.component.stateful.StatefulSessionBeanClassTable`

With that change, I get build failures:
```
[INFO] WildFly: Jipijapa SPI .............................. FAILURE [  0.263 s]
.
.
.
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project jipijapa-spi: Compilation failure: Compilation failure: 
[ERROR] /home/smarlow/work/wildfly/jpa/spi/src/main/java/org/jipijapa/plugin/spi/EntityManagerFactoryBuilder.java:[25,27] package jakarta.persistence does not exist
[ERROR] /home/smarlow/work/wildfly/jpa/spi/src/main/java/org/jipijapa/plugin/spi/EntityManagerFactoryBuilder.java:[40,5] cannot find symbol
```

**Comment by chengfang** [comment]
If we have legacy.jakarta.persistence-api in legacy-ee pom, and jakarta.persistence-api in standard-ee pom, then each consumer component module can use either one of the 2 bom. Would that work?

**Comment by scottmarlow** [comment]
@chengfang It doesn't build but perhaps that will change when we have more changes:

```
./build.sh clean install -DskipTests=true 
/home/smarlow/work/wildfly/mvnw -Dmaven.user.home=/home/smarlow/work/wildfly/tools  clean install   '-DskipTests=true'
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.wildfly:wildfly-bean-validation:jar is missing. @ line 71, column 21
[ERROR] 'dependencies.dependency.version' for org.wildfly:wildfly-ejb3:jar is missing. @ line 96, column 21
[ERROR] 'dependencies.dependency.version' for org.wildfly:wildfly-weld-common:jar is missing. @ line 133, column 21
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.wildfly:wildfly-jpa:27.0.0.Final-SNAPSHOT (/home/smarlow/work/wildfly/jpa/subsystem/pom.xml) has 3 errors
[ERROR]     'dependencies.dependency.version' for org.wildfly:wildfly-bean-validation:jar is missing. @ line 71, column 21
[ERROR]     'dependencies.dependency.version' for org.wildfly:wildfly-ejb3:jar is missing. @ line 96, column 21
[ERROR]     'dependencies.dependency.version' for org.wildfly:wildfly-weld-common:jar is missing. @ line 133, column 21
```

File: jpa/spi/pom.xml [hunk 1]
```diff
@@ -55,10 +55,16 @@
...
 
+        <dependency>
+            <groupId>jakarta.transaction</groupId>
+            <artifactId>jakarta.transaction-api</artifactId>
+        </dependency>
+
         <dependency>
             <groupId>org.jboss.spec.javax.transaction</groupId>
             <artifactId>jboss-transaction-api_1.3_spec</artifactId>
         </dependency>
```
Comment (bstansberry) [inline jpa/spi/pom.xml:66]: This should be removed.

File: jpa/subsystem/pom.xml [hunk 2]
```diff
@@ -95,12 +95,24 @@
 
         <dependency>
             <groupId>${project.groupId}</groupId>
-            <artifactId>wildfly-ejb3</artifactId>
+            <artifactId>wildfly-ejb3-jakarta</artifactId>
```
Comment (bstansberry) [inline jpa/subsystem/pom.xml:98]: This isn't needed any longer as 'wildfly-ejb3' is now the jakarta.* artifact.

File: jpa/subsystem/pom.xml [hunk 3]
```diff
@@ -95,12 +95,24 @@
...
+	    <groupId>org.jboss</groupId>
+	    <artifactId>jboss-transaction-spi-jakarta</artifactId>
+	    <version>${version.org.jboss.jboss-transaction-spi}</version>
+	    <exclusions>
+	        <exclusion>
+		    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
```
Comment (bstansberry) [inline jpa/subsystem/pom.xml:114]: Everything but the groupId and artifactId should come from dependencyManagement in the bom.

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPAInterceptorProcessor.java [hunk 4]
```diff
@@ -45,6 +45,9 @@
  */
 public class JPAInterceptorProcessor implements DeploymentUnitProcessor {
 
+    static final Class org_jboss_as_ejb3_component_session_SessionBeanComponentDescription_class =
+    JPAInterceptorProcessor.class;
```
Comment (bstansberry) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPAInterceptorProcessor.java:49]: I don't see any use of this.

File: pom.xml [hunk 5]
```diff
@@ -487,7 +487,7 @@
         <version.org.eclipse.microprofile.rest.client.api>3.0</version.org.eclipse.microprofile.rest.client.api>
         <version.org.eclipse.yasson>3.0.1</version.org.eclipse.yasson>
         <legacy.version.org.eclipse.persistence.eclipselink>2.7.11</legacy.version.org.eclipse.persistence.eclipselink>
```
Comment (bstansberry) [inline pom.xml:489]: Is this still used anywhere in the build?

File: boms/legacy-ee/pom.xml [hunk 6]
```diff
@@ -232,7 +188,7 @@
             </dependency>
             <dependency>
                 <groupId>${ee.maven.groupId}</groupId>
-                <artifactId>wildfly-weld-transactions-legacy-namespace</artifactId>
+                <artifactId>wildfly-weld-transactions</artifactId>
```
Comment (bstansberry) [inline boms/legacy-ee/pom.xml:191]: I don't think these legacy boms should include the jakarta.* dependencies. The boms only exist to support building the old javax modules.

These 'get rid of source-transform' PRs should be about removing use of the legacy boms, so the only change to them would be removing dependency entries that the build no longer needs, like you do above with some of the jipijapa ones

File: boms/legacy-ee/pom.xml [hunk 7]
```diff
@@ -243,12 +199,12 @@
             </dependency>
             <dependency>
                 <groupId>${full.maven.groupId}</groupId>
-                <artifactId>wildfly-opentelemetry-legacy-namespace</artifactId>
+                <artifactId>wildfly-opentelemetry</artifactId>
```
Comment (bstansberry) [inline boms/legacy-ee/pom.xml:202]: Same comment as above for wildfly-weld-transactions

File: boms/legacy-ee/pom.xml [hunk 8]
```diff
@@ -243,12 +199,12 @@
...
                 <version>${full.maven.version}</version>
             </dependency>
             <dependency>
                 <groupId>${full.maven.groupId}</groupId>
-                <artifactId>wildfly-opentelemetry-api-legacy-namespace</artifactId>
+                <artifactId>wildfly-opentelemetry-api</artifactId>
```
Comment (bstansberry) [inline boms/legacy-ee/pom.xml:207]: Same comment as above for wildfly-weld-transactions

File: pom.xml (hunk 5 above)
Comment (scottmarlow) [reply pom.xml:489]: Just in https://github.com/wildfly/wildfly/blob/main/boms/legacy-ee/pom.xml#L68

Comment (scottmarlow) [reply pom.xml:489]: I'll remove it

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPAInterceptorProcessor.java (hunk 4 above)
Comment (scottmarlow) [reply jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPAInterceptorProcessor.java:49]: Thanks, removed that!

File: jpa/subsystem/pom.xml (hunk 2 above)
Comment (scottmarlow) [reply jpa/subsystem/pom.xml:98]: Thanks, removed the `-jakarta`

File: jpa/subsystem/pom.xml (hunk 3 above)
Comment (scottmarlow) [reply jpa/subsystem/pom.xml:114]: I removed the wildcard excludes.

**Comment by scottmarlow** [comment]
@bstansberry this one is ready

**Comment by scottmarlow** [comment]
I just synced with the latest changes.  I'll squash the changes on the next sync.

File: jpa/spi/pom.xml [hunk 9]
```diff
@@ -53,12 +53,14 @@
         <dependency>
             <groupId>jakarta.persistence</groupId>
             <artifactId>jakarta.persistence-api</artifactId>
+            <version>${version.jakarta.persistence}</version>
```
Comment (bstansberry) [inline jpa/spi/pom.xml:56]: This isn't needed; it's controlled by the bom.

File: jpa/pom.xml [hunk 10]
```diff
@@ -39,25 +39,12 @@
...
-                <!-- TODO once moved to jakarta.* move to main module set-->
-                <module>spi</module>
-                <!-- TODO once moved to jakarta.* move to main module set-->
-                <module>eclipselink</module>
-            </modules>
-        </profile>
-    </profiles>
+        <module>subsystem</module>
+        <module>spi</module>
+<!-- TODO enable eclipselink module after resolving
```
Comment (bstansberry) [inline jpa/pom.xml:44]: @scottmarlow Is there a JIRA to track doing this?

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: jpa/spi/pom.xml (hunk 9 above)
Comment (scottmarlow) [reply jpa/spi/pom.xml]: Removed.

File: jpa/pom.xml (hunk 10 above)
Comment (scottmarlow) [reply jpa/pom.xml]: Added commit for EclipseLink, the rest of the changes are squashed into other commit.

Comment (scottmarlow) [reply jpa/pom.xml]: I look forward to creating the separate project to host feature packs for adding EclipseLink and possibly some other persistence providers.

Comment (scottmarlow) [reply jpa/pom.xml]: Sorry, the jira is https://issues.redhat.com/browse/WFLY-17123

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Pull request merged**